### DISCUSSION TITLE
Please bring back support for win2d

### DISCUSSION THREAD

**Discussion by jtorjo** [microsoft/WindowsAppSDK#435] [General]
We used the have @shawnhar and @benstevens48 answer questions on github whenever people would have questions.

Now, there's nothing. win2d is certainly not easy to use on non-trivial scenarios, and right now there's no help. Since Dec 2 2020, there are no answers to win2d questions on github.

**Comment by benstevens48** [comment]
Hi @jtorjo, I'm not sure you meant to mention me, because I am not a Microsoft employee, but I have contributed to Win2D in the past. Yes, I agree that the complete lack of engagement by anyone from Microsoft in Win2D for two months does make me quite worried about the future of Win2D. I mean it has been in a sort of maintenance mode for a while, but Shawn Hargreaves was quite good at responding to questions. 

I am sort of interested in contributing to Win2D as a project, although it depends on time and is likely to be limited to stuff that will help my app. I have a little understanding of some of the internals since I added two effects in a PR. I also have 3 open pull requests, two for a deadlock fix in interop scenarios, and one for better reference cleanup after using controls. I currently use my own fork of Win2D because I need these fixes.

Unfortunately I don't think I have a deep enough knowledge to answer many questions about Win2D without spending a long time investigating myself.

**Reply by jtorjo** [reply]
@benstevens48 Oh wow, I had no idea you don't work for Microsoft :)  I remember seeing a lot of answers from you.

> Yes, I agree that the complete lack of engagement by anyone from Microsoft in Win2D for two months does make me quite worried about the future of Win2D

Yeah, it's sad news all over. This should be the cornerstone of doing DirectX in c#/UWP -- since basically there's no other wrapper lib that comes even close. 

Not to mention that I asked for a port to WinUI Desktop more than 1.5 years ago.

> Unfortunately I don't think I have a deep enough knowledge to answer many questions about Win2D without spending a long time investigating myself.

Yeah, I totally understand. I tried to look at the code myself, but couldn't understand much of it, and I also don't have time to deeply delve into it.

**Comment by myokeeh** [comment]
I need Win2D. Keep it alive.

**Comment by duncanmacmichael** [comment]
Hi @jtorjo, thanks for opening this issue! We're sorry for the delay in response, but rest assured Win2D is alive and well and we are absolutely still working on it. :) Ownership just changed hands internally, which is why we haven't been as responsive as before.

In the short term, we're updating Win2D to work with the upcoming release of WinUI 3. In the long term, we're figuring out the right strategy for Win2D that will allow us to efficiently update it for future usage. Stay tuned and thanks again!

**Reply by jtorjo** [reply]
Hi @duncanmacmichael ,
That is awesome news!
Just to make sure: are you porting win2d to work for WinUI 3 Desktop? That would be beyond insane!

**Reply by duncanmacmichael** [reply]
Hi @jtorjo, yes that is correct - WinUI 3 Desktop is our current target. Glad you're excited about it!

**Reply by jtorjo** [reply]
@duncanmacmichael Awesome!!!
To say that I'm really excited about this is an understatement! Can't wait to have this working!

**Reply by jschroedl** [reply]
We are super excited about using win2d in a WinUI desktop app (c++). This makes me very happy to read about continued support.
